charts/stable/nextcloud/templates/_configmap.tpl

```diff
@@ -184,6 +184,9 @@ nextcloud-config:
     NX_ONLYOFFICE: {{ .Values.nextcloud.onlyoffice.enabled | quote }}
     {{- if .Values.nextcloud.onlyoffice.enabled }}
     NX_ONLYOFFICE_URL: {{ .Values.nextcloud.onlyoffice.url | quote }}
+    NX_ONLYOFFICE_INTERNAL_URL: {{ .Values.nextcloud.onlyoffice.internal_url | quote }}
+    NX_ONLYOFFICE_VERIFY_SSL: {{ .Values.nextcloud.onlyoffice.verify_ssl | quote }}
+    NX_ONLYOFFICE_NEXTCLOUD_INTERNAL_URL: {{ printf "http://%v:%v" $fullname .Values.service.main.ports.main.port }}
```

Comment:
Are the internal URLs required to work? or are those optional?

Should we "fail" on the initializtion (on the container entrypoint) if those are unset?

Reply:
Those are under the "advanced" configuration of the app in the UI. So I belive they are optional. I assume, if not set, the public FQDN of Nextcloud and the configured public FQDN of OnlyOffice is used for communication between the two as well. 

However, on SCALE they are running inside the cluster and are not going to be able to resolve those public FQDNs from inside the cluster unless the following conditions are met:
- k8s-gateway is configured
- the user has set up their router to bounce DNS requests for their root domain back to k8s gateway
- both apps accept connections to their respective parts of the API from the user's local network.

So in summary they are optional but it would almost certainly not work out of the box if they are not set.

Reply:
On second thought we should actually require this, because otherwise it would overwrite whatever the user has set in the UI with the default value in the chart ("") which would be a breaking change.

Reply:
Let’s require in both sides